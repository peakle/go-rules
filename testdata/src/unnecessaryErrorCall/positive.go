package unnecessaryErrorCall

import (
	"errors"
	"fmt"
)

func foo() error {
	fmt.Println(errors.New(fmt.Sprintf("foo: %s", "bar")))            // want `use fmt.Errorf instead of nested calls`
	fmt.Println(errors.New(fmt.Sprintf("foo: %s, %s", "bar", "boo"))) // want `use fmt.Errorf instead of nested calls`

	return errors.New(fmt.Sprintf("foo: %s", "bar")) // want `use fmt.Errorf instead of nested calls`
}
