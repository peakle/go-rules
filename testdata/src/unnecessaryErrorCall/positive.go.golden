package unnecessaryErrorCall

import (
	"errors"
	"fmt"
)

func foo() error {
	fmt.Println(fmt.Errorf("foo: %s", "bar"))            // want `use fmt.Errorf instead of nested calls`
	fmt.Println(fmt.Errorf("foo: %s, %s", "bar", "boo")) // want `use fmt.Errorf instead of nested calls`

	return fmt.Errorf("foo: %s", "bar") // want `use fmt.Errorf instead of nested calls`
}
